scripts/change-log-generator-utils.js

```diff
@@ -0,0 +1,274 @@
+const RELEASE_DATE_ARG = '-t';
+const PACKAGES = 'PACKAGES';
+const TYPE_HEADER = '\n## %s\n';
+const SECTION_HEADER = '\n#### %s\n';
+const MESSAGE_FORMAT =
+  '\n- %s ([PR #%s](https://github.com/forcedotcom/salesforcedx-vscode/pull/%s))\n';
+const PR_ALREADY_EXISTS_ERROR =
+  'Filtered PR number %s. An entry already exists in the changelog.';
+
+const typesToIgnore = [
+  'chore',
+  'style',
+  'refactor',
+  'test',
+  'build',
+  'ci',
+  'revert'
+];
+
+module.exports = {
+  getPreviousReleaseBranch, parseCommits, getMessagesGroupedByPackage, getChangeLogText, getCommits, writeChangeLog
+}
+
+/**
+ *  Returns the previous release branch
+ */
+function getPreviousReleaseBranch(releaseBranch) {
+  const releaseBranches = getReleaseBranches();
+  return releaseBranches[0];
+}
+
+/**
+ * Returns a list of remote release branches, sorted in reverse order by
+ * creation date. This ensures that the first entry is the latest branch.
+ */
+ function getReleaseBranches() {
+  return shell
+    .exec(
+      `git branch --remotes --list --sort='-creatordate' '${constants.RELEASE_BRANCH_PREFIX}*'`,
+      { silent: false }
+    )
+    .replace(/\n/g, ',')
+    .split(',')
+    .map(Function.prototype.call, String.prototype.trim);
+}
+
+/**
+ * This command will list all commits that are different between
+ * the two branches. Therefore, we are guaranteed to get all new
+ * commits relevant only to the new branch.
+ */
+ function getCommits(releaseBranch, previousBranch) {
+  const commits = shell
+    .exec(
+      `git log --cherry-pick --oneline ${releaseBranch}...${previousBranch}`,
+      {
+        silent: false
+      }
+    )
+    .stdout.trim()
+    .split('\n');
+  return commits;
+}
+
+/**
+ * Parse the commits and return them as a list of hashmaps.
+ */
+ function parseCommits(commits) {
+  let commitMaps = [];
+  for (let i = 0; i < commits.length; i++) {
+    const commitMap = buildMapFromCommit(commits[i]);
+    if (commitMap && Object.keys(commitMap).length > 0) {
+      commitMaps.push(commitMap);
+    }
+  }
+  return filterExistingPREntries(commitMaps);
+}
+
+function buildMapFromCommit(commit) {
+  let map = {};
+  if (commit) {
+    let pr = constants.PR_REGEX.exec(commit);
+    let commitNum = constants.COMMIT_REGEX.exec(commit);
+    if (pr && commitNum) {
+      let message = commit.replace(commitNum[0], '').replace(pr[0], '');
+      let type = constants.TYPE_REGEX.exec(message);
+      map[PR_NUM] = pr[0].replace(/[^\d]/g, '');
+      map[COMMIT] = commitNum[0];
+      if (type) {
+        map[TYPE] = type[1];
+        message = message.replace(type[0], '');
+      }
+      message = message.trim();
+      map[MESSAGE] = message.charAt(0).toUpperCase() + message.slice(1);
+      map[FILES_CHANGED] = getFilesChanged(map[COMMIT]);
+      map[PACKAGES] = getPackageHeaders(map[FILES_CHANGED]);
+    }
+  }
+
+  console.log('\nCommit: ' + commit);
+  console.log('Commit Map:');
+  console.log(map);
```

Comment:
did you purposely leave these consoles here?

Reply:
I moved most of these functions from changelog generator and this is something I want to leave it as is.